### PR TITLE
fix: protect external bandada apis

### LINKED ISSUE #899: fix: refactorig Bandada Provider APIs

### PR DESCRIPTION
**Pull request by 0xisk**
## Explanation

In this PR changes includes protecting the group Bandada APIs. 

## Related Issues

- Fixes #899 

## Screenshots

N/A

## Manual Testing Steps

- Please check Bandada function from the Demo.

## Pre-Merge Checklist

- [x] PR template is filled out
- [x] Pre-commit and pre-push hook checks are passed
- [x] E2E tests are passed locally
- [x] **IF** this PR fixes a bug, a test that _would have_ caught the bug has been added
- [x] PR is linked to the appropriate GitHub issue
- [x] PR has been added to the appropriate release Milestone

> PR template source from [github.com/MetaMask](https://github.com/MetaMask)

### CODE REVIEW

File: packages/app/src/background/services/validation/source.ts [hunk 1]
```diff
@@ -15,18 +15,15 @@ export const validateMerkleProofSource = ({
...
     const merkleProofArtifacts: IMerkleProofArtifacts = merkleProofSource as IMerkleProofArtifacts;
     return { merkleProofArtifacts };
   }
-
   if (isMerkleProof(merkleProofSource)) {
     const merkleProofProvided: IMerkleProof = merkleProofSource as IMerkleProof;
     new MerkleProofValidator(merkleProofProvided).validateProof();
     return { merkleProofProvided };
   }
-
```
Comment (0xmad) [inline packages/app/src/background/services/validation/source.ts:29]: Please keep new lines

File: packages/providers/src/sdk/CryptKeeperInjectedProvider.ts [hunk 2]
```diff
@@ -87,7 +87,7 @@ export class CryptKeeperInjectedProvider extends Handler implements ICryptKeeper
 
   async joinGroup({ groupId, apiKey, inviteCode }: IJoinGroupMemberArgs): Promise<void> {
     await this.post({
-      method: RPCExternalAction.JOIN_GROUP_REQUEST,
+      method: RPCExternalAction.INJECTOR_JOIN_GROUP_REQUEST,
```
[the remark this answers is not in the document]
Comment (0xisk) [reply packages/providers/src/sdk/CryptKeeperInjectedProvider.ts:90]: That would means its an external RPC executed via the injector contoller.

[the remark this answers is not in the document]
Comment (0xmad) [reply packages/providers/src/sdk/CryptKeeperInjectedProvider.ts:90]: Injector is an entry point for all external communication. I think adding this prefix for key name is redundant. It's useful for enum values not for keys.

**Review by 0xmad — APPROVED** [review]
@0xisk thanks

**Pull request merged**